classylss/binding.pyx

```diff
@@ -666,62 +684,66 @@ cdef class Background:
         Density parameter of relativistic (radiation like) component, including
         relativistic part of massive neutrino and massless neutrino.
         """
-        return self.compute_for_z(z, self.ba.index_bg_Omega_r)
+        E = self.efunc(z)
+        z = np.array(z, dtype=np.float64)
+        return self.Omega0_r * (1+z)**4 / E**2
 
     def Omega_m(self, z):
         """
         Density parameter of non-relativistic (matter like) component, including
         non-relativistic part of massive neutrino. Unit
         """
-        return self.compute_for_z(z, self.ba.index_bg_Omega_m)
+        E = self.efunc(z)
+        z = np.array(z, dtype=np.float64)
+        return self.Omega0_m * (1+z)**3 / E**2
```

Comment:
Same reason as above, matter contains a neutrino part that is redshift dependent, so you can't scale it from today's value.